lib/game.rb

```diff
@@ -0,0 +1,86 @@
+require "yaml"
+require_relative "validate"
+require_relative "show_content"
+require_relative "db_methods"
+require_relative "user"
+module NewSuperCodebreaker2021
+  class Game
+    include Validate
+    include ShowContent
+    include DBMethods
+
+    START_COMMANDS = %i[start rules stats exit].freeze
+    DIFFICULTY_COMMANDS = %i[easy medium hell exit].freeze
+    GUESS_COMMANDS = %i[hint rules exit].freeze
+    AFTER_GAME_COMMANDS = %i[start save exit].freeze
+    YES_NO_COMMANDS = %i[yes no].freeze
+
+    def chose_com(command)
+      START_COMMANDS.include?(command.to_sym) ? command.to_sym : false
+    end
+
+    def take_name(input_name)
+      if input_name == "exit"
+        :exit
+      else
+        validate_name(input_name)
+      end
+    end
+
+    def chose_difficulty(difficulty)
+      DIFFICULTY_COMMANDS.include?(difficulty.to_sym) ? difficulty.to_sym : false
+    end
+
+    def generate_code
+      Array.new(4) { rand(1..6) }
+    end
+
+    def user_guess(code)
+      GUESS_COMMANDS.include?(code.to_s.to_sym) ? code.to_s.to_sym : validate_user_code(code)
+    end
+
+    def take_hint(user, code, used_hints)
```

Comment:
From where do we get the `used_hints`?

Reply:
code.sample is our used_hint, so because of returning this value we can send it to method